gtsam/nonlinear/NonlinearEquality.h

```diff
@@ -158,11 +160,20 @@ class NonlinearEquality: public NoiseModelFactorN<VALUE> {
     }
   }
 
+  Vector unwhitenedError(const Values& x, OptionalMatrixVecType H = nullptr) const override {
+    VALUE x1 = x.at<VALUE>(key());
+    if (H) {
+      return evaluateError(x1, &(H->front()));
```

Comment:
does the base class not have unwhitenedError? Why not? Would that not be easier?

Reply:
The parent classes have unwhitenedError as pure virtual function, because different types of constraint representation have different ways of computing the unwhitenedError. e.g., ExpressionEqualityConstraint computes unwhitenedError by evaluating an expression; ZeroCostConstraint computes unwhitenedError by evaluating the error vector of the underlying factor. NonlinearEquality (refactored from previous code) represents a specific type of constraint with its specific evaluateError function which should be called to compute unwhitenedError.